orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTaskTest.java

```diff
@@ -0,0 +1,211 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Header;
+import retrofit.client.OkClient;
+import retrofit.client.Response;
+import retrofit.converter.ConversionException;
+import retrofit.converter.Converter;
+import retrofit.converter.JacksonConverter;
+import retrofit.mime.TypedInput;
+
+public class MonitoredDeployBaseTaskTest {
+
+  private MonitoredDeployBaseTask monitoredDeployBaseTask;
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @BeforeEach
+  void setup() {
+    OkClient okClient = new OkClient();
+    RestAdapter.LogLevel retrofitLogLevel = RestAdapter.LogLevel.NONE;
+
+    RequestInterceptor requestInterceptor = request -> {};
+    DeploymentMonitorDefinition deploymentMonitorDefinition = new DeploymentMonitorDefinition();
+    deploymentMonitorDefinition.setId("LogMonitorId");
+    deploymentMonitorDefinition.setName("LogMonitor");
+    deploymentMonitorDefinition.setFailOnError(true);
+    var deploymentMonitorDefinitions = new ArrayList<DeploymentMonitorDefinition>();
+    deploymentMonitorDefinitions.add(deploymentMonitorDefinition);
+
+    DeploymentMonitorServiceProvider deploymentMonitorServiceProvider =
+        new DeploymentMonitorServiceProvider(
+            okClient, retrofitLogLevel, requestInterceptor, deploymentMonitorDefinitions);
+    monitoredDeployBaseTask =
+        new MonitoredDeployBaseTask(deploymentMonitorServiceProvider, new NoopRegistry());
+  }
+
+  @Test
+  public void shouldParseHttpErrorResponseDetailsWhenHttpErrorHasOccurred() {
+
+    var converter = new JacksonConverter(objectMapper);
+    var responseBody = new HashMap<String, String>();
+    var headers = new ArrayList<Header>();
+    var header = new Header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+
+    headers.add(header);
+    responseBody.put("error", "404 - Resource not found");
+
+    Response response =
+        new Response(
+            "/deployment/evaluateHealth",
+            HttpStatus.BAD_REQUEST.value(),
+            HttpStatus.BAD_REQUEST.name(),
+            headers,
+            new MockTypedInput(converter, responseBody));
+
+    RetrofitError httpError =
+        RetrofitError.httpError(
+            "https://foo.com/deployment/evaluateHealth", response, new JacksonConverter(), null);
+
+    String logMessageOnHttpError =
+        monitoredDeployBaseTask.getRetrofitLogMessage(httpError.getResponse());
+    String status = HttpStatus.BAD_REQUEST.value() + " (" + HttpStatus.BAD_REQUEST.name() + ")";
+    String body = "{\"error\":\"404 - Resource not found\"}";
+
+    assertEquals(
```

Comment:
I'd like to start using `org.assertj.core.api.Assertions.assertThat` instead of the jupiter assertions.  It not in this PR, then let's do it going forward.